File: python-bot/handler.py
from datetime import datetime

# Configuration variables
name = "Python"
avatar = "https://i.pinimg.com/736x/28/17/e1/2817e11cb843de716180346af96b0a0b.jpg"
port = 8080

def handle_request(data):
    """
    Function to process the incoming request and return a response.
    This can be modified by other developers.
    data has key 'messages' with a list of messages:
    - {'role': ..., 'content': text of the message}

    :param data: Parsed JSON request data.
    :return: Dictionary representing the response
    """
    print(f'{data}')

    return {
        "model": "echo",
        "done": True,
        "created_at": datetime.utcnow().isoformat(),
        "message": {
            "role": "user",
            "content": "new content generated in python"
        }
    }

File: bots/ruby/handler.rb
module Handler
  NAME = "Ruby"
  AVATAR = "https://cdn.dribbble.com/userupload/21266107/file/original-e64fa9cba72c8d48ca14c6978a67fb2f.png"
  PORT = 8020
  SYSTEM = "itsa me ruby."

  def self.handle_request(data)
    puts "Received data: #{data}"
    {
      model: "echo",
      done: true,
      created_at: Time.now.utc.iso8601,
      message: {
        role: "user",
        content: "new content generated in ruby"
      }
    }
  end
end
